main/lsp/requests/document_formatting.cc

```diff
@@ -16,9 +16,38 @@ LSPTask::Phase DocumentFormattingTask::finalPhase() const {
     return Phase::INDEX;
 }
 
+enum RubyfmtStatus {
+    OK = 0,
+
+    // passed buffer contained a ruby syntax error. Non fatal, user should feel
+    // free to continue to call rubyfmt with non-error strings.
+    SYNTAX_ERROR = 1,
+
+    // this error is fatal, the calling program should not continue to execute
+    // rubyfmt and you should report a bug with the file that crashed rubyfmt
+    RIPPER_PARSE_FAILURE = 2,
+
+    // an error occured during IO within the function, should be impossible
+    // and most likely indicates a programming error within rubyfmt, please
+    // file a bug
+    IO_ERROR = 3,
+
+    // some unknown ruby error occured during execution fo Rubyfmt. This indicates
+    // a programming error. Please file a bug report and terminate the process
+    // and restart.
+    OTHER_RUBY_ERROR = 4,
+
+    // Rubyformat failed to initialize.
+    // N.B. 101 is the default error code for _any_ Rust panic,
+    // but a failure to initialize is the most probable cause for this.
+    INITIALIZE_ERROR = 101,
+
+    RUBYFMT_NOT_IN_PATH = 127,
+};
+
 void DocumentFormattingTask::index(LSPIndexer &index) {
     auto response = make_unique<ResponseMessage>("2.0", id, LSPMethod::TextDocumentFormatting);
-    if (!config.opts.lspDocumentFormatRubyfmtEnabled) {
+    if (!(config.opts.lspDocumentFormatRubyfmtEnabled && rubyfmt_enabled)) {
```

Comment:
Either in this PR or #5834, can we also delete the config `RUBYFMT_ENABLED` macro, the `rubyfmt_enabled` constexpr bool, and the `build:rubyfmt` config?

Reply:
Will do in #5834 👍